spec/helpers/ajax-helper.ts

```diff
@@ -139,7 +139,7 @@ export class MockXMLHttpRequest {
   onerror: (e: ErrorEvent) => any;
   onprogress: (e: ProgressEvent) => any;
   ontimeout: (e: ProgressEvent) => any;
-  upload: XMLHttpRequestUpload;
+  upload: XMLHttpRequestUpload = Object.create(null);
```

Comment:
what about simply do 

```js
upload = {
    progress: {
      get() {
        return this.upload.onprogress;
      }
    }
  }
```

? any strong reason to declare empty object then define property later?
(other than specifying type `XMLHttpRequestUpload`. This'll be easily solvable with TS2.1 by declare as `Partial<XMLHttpRequestUpload>`)